### PR TITLE
Fix#1254#1305 bug on JSON serialization and deserialization

### PR DESCRIPTION
**Pull request by gofow**
### Motivation:

Fix#1254#1305 bug on JSON serialization and deserialization

### Modification:

1. In BeanSerializer.serialize(), create a fresh Map when the "bean" is a Map instance, in order to prevent any unpredictable changes due to references
2. In BeanSerializer.serialize(), use class.getName() rather than class.getCanonicalName() when the bean is a custom object, in order to get correct representation of inner class for class loader
3. Add two cases to JSONTest

### CODE REVIEW

**Comment by EvenLjj** [comment]
image
@gofow Pls format code by `mvn clean compile` before git push.

**Comment by gofow** [comment]
@EvenLjj Thank you! Actually I'm feeling confused about the error, and I have done with the format.

File: core/common/src/test/java/com/alipay/sofa/rpc/common/json/JSONTest.java [hunk 1]
```diff
@@ -38,4 +41,23 @@ public void test() {
         Assert.assertTrue(str.contains(JSON.CLASS_KEY));
     }
 
+    @Test
+    public void BeanWithMapOriginalObjectTest() {
```
Comment (EvenLjj) [inline core/common/src/test/java/com/alipay/sofa/rpc/common/json/JSONTest.java:45]: Pay a little attention to the method format.

Comment (gofow) [reply core/common/src/test/java/com/alipay/sofa/rpc/common/json/JSONTest.java:45]: You're right. It looks like a class name right now... Maybe a name like testBeanWithMapSerialization() will be better, and I should look around methods in other test class before.

Comment (EvenLjj) [reply core/common/src/test/java/com/alipay/sofa/rpc/common/json/JSONTest.java:45]: Yes, it's better to modify it, there are no other problems about this PR. And we will submit another PR to fix the single test problem `testCustomThrowableDeserializerEnabled`.

Comment (gofow) [reply core/common/src/test/java/com/alipay/sofa/rpc/common/json/JSONTest.java]: Glad to hear that. And it is strange that this test passed on my PC.

Comment (EvenLjj) [reply core/common/src/test/java/com/alipay/sofa/rpc/common/json/JSONTest.java]: This is usually caused when the unit tests run concurrently.

**Review by EvenLjj — APPROVED** [review]
LGTM

**Review by Lo1nt — APPROVED** [review]

**Pull request merged**